File: build-runner.py
import subprocess, traceback
try:
    r = subprocess.run(
        ['node', 'node_modules/vite/bin/vite.js', 'build'],
        capture_output=True, text=True, cwd='C:\\Users\\User\\Desktop\\likelink2',
        timeout=120
    )
    content = "STDOUT:\n" + r.stdout + "\nSTDERR:\n" + r.stderr + "\nEXIT CODE: " + str(r.returncode)
    open('C:\\Users\\User\\Desktop\\likelink2\\build-result.txt', 'w').write(content)
    print("Build completed with exit code:", r.returncode)
except Exception as e:
    content = "EXCEPTION: " + str(e) + "\n\nTRACEBACK:\n" + traceback.format_exc()
    open('C:\\Users\\User\\Desktop\\likelink2\\build-result.txt', 'w').write(content)
    print("Build failed with exception:", e)

File: build-runner.js
import { execFileSync } from 'node:child_process';
import { writeFileSync } from 'node:fs';

const projectRoot = 'C:\\Users\\User\\Desktop\\likelink2';
const outputFile = 'C:\\Users\\User\\Desktop\\likelink2\\build-result.txt';
const nodePath = 'C:\\Program Files\\nodejs\\node.exe';

writeFileSync(outputFile, 'STARTED - node at: ' + nodePath);

try {
  const stdout = execFileSync(nodePath, ['node_modules/vite/bin/vite.js', 'build'], {
    cwd: projectRoot,
    encoding: 'utf8',
    timeout: 120000,
    stdio: ['pipe', 'pipe', 'pipe']
  });
  writeFileSync(outputFile, 'BUILD_SUCCESS\n' + stdout);
} catch (e) {
  const err = e.stderr || e.stdout || e.message || 'Unknown error';
    writeFileSync(outputFile, 'BUILD_FAILED\n' + err + '\nExit code: ' + e.status + '\nSignal: ' + e.signal);
}